src/modules/file/file.repository.ts

```diff
@@ -292,6 +304,53 @@ export class SequelizeFileRepository implements FileRepository {
     return files.map(this.toDomain.bind(this));
   }
 
+  async getSumSizeOfFilesByStatuses(
+    createdBy: WorkspaceItemUserAttributes['createdBy'],
+    workspaceId: WorkspaceAttributes['id'],
+    statuses: FileStatus[],
+    options?: {
+      createdFrom?: Date;
+      removedFrom?: Date;
+      limit: number;
+      offset: number;
+      order?: Array<[keyof File, string]>;
+    },
+  ): Promise<{ size: string }[]> {
+    const statusesFilter = statuses.map((value) => ({ status: value }));
+
+    const fileDateFilters: any = {};
+    if (options?.removedFrom) {
+      fileDateFilters.removed_at = {
+        [Op.gte]: options?.removedFrom,
+      };
+    }
+
+    const workspaceItemDateFilters: any = {};
+    if (options?.createdFrom) {
+      workspaceItemDateFilters.created_at = {
+        [Op.gte]: options.createdFrom,
+      };
+    }
+
+    const sizes = await this.fileModel.findAll({
+      attributes: ['size'],
+      limit: options?.limit,
+      offset: options?.offset,
+      where: {
+        [Op.or]: statusesFilter,
+        ...fileDateFilters,
+      },
+      include: {
+        model: WorkspaceItemUserModel,
+        attributes: [],
+        where: { createdBy, workspaceId, ...workspaceItemDateFilters },
+      },
+      order: options?.order,
+    });
+
+    return sizes as unknown as { size: string }[];
+  }
```

Comment:
I wanted to use .sum or try to get the total using the raw:true option with sequelize.literal inside the attributes field. 
However, sequelize is not very happy when using includes, group by and a SUM inside attributes at the same time. It continues asking to group by fields that just makes the query complex, harder to read and it still returns an array.

The only way to get the sum directly from the DB is probably using a raw query, but in this case, I preferred not to use that approach and just use .reduce to sum the values.